### PR TITLE
Address Mqtt source review feedback

### PR DESCRIPTION
**Pull request by 2m**
* remove race condition between connect and message handler set
* final case classes
* make handlers abstract, instead of asyncCallback getters
* make internal implementation private to the package
* abstract over implicit argument lists

### CODE REVIEW

File: mqtt/src/main/scala/Mqtt.scala [hunk 1]
```diff
@@ -9,27 +9,31 @@ import org.eclipse.paho.client.mqttv3.{ MqttMessage => PahoMqttMessage, _ }
...
   broker:      String,
   clientId:    String,
   persistence: MqttClientPersistence
 )
 
-case class MqttMessage(topic: String, payload: ByteString)
+final case class MqttMessage(topic: String, payload: ByteString)
 
-trait MqttConnectorLogic { this: GraphStageLogic =>
+private[mqtt] trait MqttConnectorLogic { this: GraphStageLogic =>
```
Comment (johanandren) [inline mqtt/src/main/scala/Mqtt.scala]: `/** Internal API`

File: mqtt/src/main/scala/Mqtt.scala [hunk 2]
```diff
@@ -49,6 +52,7 @@ trait MqttConnectorLogic { this: GraphStageLogic =>
       def connectionLost(cause: Throwable) =
         onConnectionLost.invoke(cause)
     })
+    client.connect((), connectHandler(client))
```
Comment (johanandren) [inline mqtt/src/main/scala/Mqtt.scala]: 👍

File: mqtt/src/test/scala/MqttSourceSpec.scala [hunk 3]
```diff
@@ -76,10 +87,22 @@ class MqttSourceSpec extends WordSpec with Matchers with ScalaFutures {
     val server = new Server()
     server.startServer()
     try {
-      test(settings)(server)(sys)(mat)
+      test(FixtureParam(settings, server, sys, mat))
     } finally {
       server.stopServer()
     }
   }
```
Comment (johanandren) [inline mqtt/src/test/scala/MqttSourceSpec.scala]: Who stops the `ActorSystem`?

**Comment by johanandren** [comment]
Lookin' good!

File: mqtt/src/main/scala/Mqtt.scala [hunk 4]
```diff
@@ -9,27 +9,31 @@ import org.eclipse.paho.client.mqttv3.{ MqttMessage => PahoMqttMessage, _ }
...
-  def onConnect: AsyncCallback[IMqttAsyncClient]
-  def onMessage: AsyncCallback[MqttMessage]
-  def onConnectionLost: AsyncCallback[Throwable]
+  def handleConnection(client: IMqttAsyncClient): Unit
+  def handleMessage(message: MqttMessage): Unit
+  def handleConnectionLost(ex: Throwable): Unit
+
+  def onConnect = getAsyncCallback[IMqttAsyncClient](handleConnection)
+  val onMessage = getAsyncCallback[MqttMessage](handleMessage)
+  val onConnectionLost = getAsyncCallback[Throwable](handleConnectionLost)
```
Comment (patriknw) [inline mqtt/src/main/scala/Mqtt.scala]: It's fine to create it like this in the constructor, but you are not allowed to `invoke` it before `preStart`. That is a problem if you expose the callback to the external world (via materialized value).

**Pull request merged**